CMakeLists.txt

```diff
@@ -0,0 +1,236 @@
+
+cmake_minimum_required(VERSION 3.15)
+cmake_policy(SET CMP0048 NEW) # project version
+cmake_policy(SET CMP0076 NEW) # full paths
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_modules")
+
+########################################################################
+# Project Details
+project(FreeRTOS-Plus-TCP
+        VERSION 3.1.0
+        DESCRIPTION "FreeRTOS TCP/UDP Network Layer"
+        HOMEPAGE_URL https://freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/index.html
+        LANGUAGES C)
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
```

Comment:
why not?

Reply:
Since it is in other `CMakeList.txt` files in this current directory, I added the same restriction here.

See other examples: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/search?q=if%28+%24%7BPROJECT_SOURCE_DIR%7D+STREQUAL+%24%7BPROJECT_BINARY_DIR%7D+%29